tests/math/test_multi_dispatch.py

```diff
@@ -196,6 +197,17 @@ def test_dot_autograd():
     assert fn.allclose(qml_grad(fn.dot)(x, y), x)
 
 
+def test_dot_autograd_with_scalar():
```

Comment:
Should we test it both with (x, y) and (y, x) to make sure it works both ways? And with tensorflow and torch as well?

Reply:
Added